src/meet-calendar.ts

```diff
@@ -0,0 +1,684 @@
+            <div class = "tzcF6">
+                <div class = "DPvwYc jitsi_edit_page_icon"/>
+            </div>
+        `);
+
+    newRow.append(`
+            <div class = "j3nyw">
+                <div class = "BY5aAd">
+                    <div role = "button"
+                        class = "uArJ5e UQuaGc Y5sE8d"
+                        id="jitsi_button_container">
+                        <content class = "CwaK9">
+                            <span id="jitsi_button"
+                                    class="RveJvd snByac">
+                            </span>
+                        </content>
+                    </div>
+                </div>
+            </div>
+        `);
+
+    newRow.insertBefore(neighbor);
+
+    return newRow.find("content");
+  }
+
+  /**
+   * Adds the jitsi button in buttonContainer.
+   */
+  addJitsiButton() {
+    var container = this.buttonContainer;
+    if (!container) return false;
+
+    this.description?.update(this.location);
+  }
+
+  /**
+   * The event description.
+   */
+  get description(): DescriptionWrapper {
+    if (!this.descriptionInstance)
+      this.descriptionInstance = new G2Description(this);
+    return this.descriptionInstance;
+  }
+}
+
+/**
+ * The google calendar specific implementation of the location field in the
+ * event page.
+ */
+class G2Location extends LocationWrapper {
+  private _getSelector() {
+    return $("#xLocIn input[jsname=YPqjbf][role=combobox]");
+  }
+
+  private _getLocationElement() {
+    let elem = this._getSelector();
+
+    if (elem.length === 0) {
+      // this is the case where location is not editable
+      let element = $("#xOnCal")[0];
+
+      if (!element) {
+        return undefined;
+      }
+
+      return {
+        val: () => element.innerHTML,
+      };
+    }
+
+    return elem;
+  }
+
+  /**
+   * The text from the location input field.
+   */
+  get text(): string | undefined {
+    let e = this._getLocationElement();
+    return e?.val() as any;
+  }
+
+  /**
+   * Adds text to location input.
+   * @param text
+   */
+  addLocationText(text: string) {
+    let elem = this._getSelector();
+
+    // in case this element is missing, means we cannot edit the text
+    if (elem.length === 0) return;
+
+    // Set the location if there is content
+    let locationNode = elem[0] as any; // GT BUG BUG BUG
+    if (locationNode) {
+      locationNode.focus(); // Focus needed to make a simulation of keying in.
+      elem.attr(
+        "value",
+        locationNode.value === "" ? text : locationNode.value + ", " + text
+      );
+      locationNode.dispatchEvent(getKeyboardEvent("input"));
+      // tried many combinations and cannot make it reliably working
+      // in some cases hovering over the input will make it save,
+      // otherwise text is seen in the input but is not saved after
+      // clicking save
+      window.setTimeout(function () {
+        locationNode.focus();
+        elem.val(elem.val() + " ");
```

Comment:
I tried another 5 variants of this code, my favorite being replacing `+ " "` with `|| " "`, but i can't make it any more reliable. i am wondering whether we should use the `dispatchEvent` code  at https://github.com/brave/jidesha/blob/master/calendar/meet-calendar.js#L517-L525

Reply:
addressed by later updates